src/app/services/request/request.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { Observer } from 'rxjs/Observer';
+
+@Injectable()
+export class RequestService {
+  public makeRequest(options: RequestOption): Observable<RequestReponse> {
+    return Observable.create((observer: Observer<RequestReponse>) => {
+      const xhr = new XMLHttpRequest();
```

Comment:
I would use the angular http client here https://angular.io/docs/ts/latest/guide/server-communication.html